bin_sort/bin_sort.c

```diff
@@ -0,0 +1,63 @@
+/* Following algorithm sorts the input array in ascending order
+ * Time Complexity : O(n)-> Avg case and O(n^2)-> Worst Case
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+struct bucket {
+	int count;
+	int* value;
+};
+
+int comparator(const int* first, const int* second) {
```

Comment:
This can be changed to 
```
int comparator(const int* first, const int* second) {
	int a = *first, b = *second;
	int result = a - b;
	return (0 < result) - (result < 0);
}
```